app/controllers/api/allocation_api_controller.rb

```diff
@@ -4,7 +4,9 @@ module Api
   class AllocationApiController < Api::ApiController
     def show
       render_404('Not ready for allocation') && return if allocation.nil?
-      render_404('Not allocated') && return if allocation.primary_pom_nomis_id.nil?
+
+      offender = OffenderService.get_offender(offender_number)
+      render_404('Not allocated') && return if allocation.primary_pom_nomis_id.nil? || !offender.inside_omic_policy?
```

Comment:
This line might be easier to read if rewritten as:

```suggestion
      render_404('Not allocated') && return unless allocation.active? && offender.inside_omic_policy?
```

I think this should be functionally equivalent (the tests should fail if it isn't!), however it does a couple of things:
- Uses the method `allocation.active?` which does exactly what you need, but is easier to read than `allocation.primary_pom_nomis_id.nil?`
- Reorders the `if` statement to become an `unless` – which in this case seems to read more clearly

(In my opinion) it makes the intention of this line clearer to understand:
> Return 404, unless the offender has an active allocation and is inside OMIC policy